DKGRituals/src/mappingCoordinator.ts

```diff
@@ -0,0 +1,137 @@
+  transaction.save();
+
+  const ritual = getOrCreateRitual(event.params.ritualId.toString());
+  ritual.initiator = ritualInst.getInitiator();
+  ritual.initTimestamp = ritualInst.getInitTimestamp();
+  ritual.endTimestamp = ritualInst.getEndTimestamp();
+  ritual.authority = event.params.authority;
+  ritual.dkgSize = ritualInst.getDkgSize();
+  ritual.threshold = ritualInst.getThreshold();
+  ritual.accessController = ritualInst.getAccessController();
+  ritual.participants = changetype<Bytes[]>(event.params.participants);
+  ritual.dkgStatus = "DKG_AWAITING_TRANSCRIPTS";
+
+  const transactions = ritual.transactions;
+  transactions.push(transaction.id);
+  ritual.transactions = transactions;
+  ritual.save();
+
+  const ritualCounter = getOrCreateRitualCounter(RITUAL_COUNTER_ID);
+  ritualCounter.total = ritualCounter.total + 1;
+  ritualCounter.notEnded = ritualCounter.notEnded + 1;
+  ritualCounter.save();
+
+  const user = getOrCreateUser(ritual.initiator);
+  const rituals = user.rituals;
+  rituals.push(ritual.id);
+  user.rituals = rituals;
+  user.save();
+}
+
+export function handleTranscriptPosted(event: TranscriptPostedEvent): void {
+  const ritual = getOrCreateRitual(event.params.ritualId.toString());
+  const postedTranscripts = ritual.postedTranscripts;
+  postedTranscripts.push(event.params.node);
+  ritual.postedTranscripts = postedTranscripts;
+  ritual.postedTranscriptsAmount = ritual.postedTranscriptsAmount + 1;
+  ritual.save();
+}
+
+export function handleStartAggregationRound(
+  event: StartAggregationRoundEvent,
+): void {
+  const transaction = getOrCreateTransaction(getTransactionIDFromEvent(event));
+  transaction.txHash = event.transaction.hash;
+  transaction.timestamp = event.block.timestamp;
+  transaction.from = event.transaction.from;
+  transaction.to = event.transaction.to;
+  transaction.description = "Posted Transcripts";
+  transaction.save();
+
+  const ritual = getOrCreateRitual(event.params.ritualId.toString());
+  ritual.dkgStatus = "DKG_AWAITING_AGGREGATIONS";
+
+  const transactions = ritual.transactions;
+  transactions.push(transaction.id);
+  ritual.transactions = transactions;
+  ritual.save();
+}
+
+export function handleAggregationPosted(event: AggregationPostedEvent): void {
+  const ritual = getOrCreateRitual(event.params.ritualId.toString());
+  const postedAggregations = ritual.postedAggregations;
+  postedAggregations.push(event.params.node);
+  ritual.postedAggregations = postedAggregations;
+  ritual.postedAggregationsAmount = ritual.postedAggregationsAmount + 1;
+  ritual.save();
+}
+
+export function handleEndRitual(event: EndRitualEvent): void {
+  const transaction = getOrCreateTransaction(getTransactionIDFromEvent(event));
+  transaction.txHash = event.transaction.hash;
+  transaction.timestamp = event.block.timestamp;
+  transaction.from = event.transaction.from;
+  transaction.to = event.transaction.to;
+  transaction.description = "Posted Aggregations";
```

Comment:
Same comment for `EndRitual` as `StartAggregationRound`.

Do we need to store the associated transaction? Presumably it's the same tx as the last `AggregationPosted` tx - https://github.com/nucypher/nucypher-contracts/blob/main/contracts/contracts/coordination/Coordinator.sol#L394 and https://github.com/nucypher/nucypher-contracts/blob/main/contracts/contracts/coordination/Coordinator.sol#L407

Reply:
The tx of `AggregationPosted` is not being stored, as pointed out on above, the transaction details are only being stored in `Transaction` entity specifically on events that updates the Ritual entity status, maybe worth storing all the transactions, and only add to the Ritual transactions property only those that are relevant for changing the Ritual status?